project/src/main/java/com/google/sps/utilities/DatabaseWrapper.java

```diff
@@ -102,10 +102,31 @@ public void insertVolunteeringOpportunity(VolunteeringOpportunity opportunity) {
     spanner.close();
   }
 
+  /**
+   * Given a volunteering opportunity, update a row with the same opportunityId in the DB
+   *
+   * @param opportunity the volunteering opportunity to be updated
+   */
+  public void updateVolunteeringOpportunity(VolunteeringOpportunity opportunity) {
+    SpannerOptions options = SpannerOptions.newBuilder().build();
+    Spanner spanner = options.getService();
+    DatabaseId db = DatabaseId.of(options.getProjectId(), instanceId, databaseId);
+    DatabaseClient dbClient = spanner.getDatabaseClient(db);
+
+    List<Mutation> mutations =
+        getMutationsFromBuilder(newUpdateBuilderFromVolunteeringOpportunity(), opportunity);
+    dbClient.write(mutations);
+    spanner.close();
+  }
+
   private static Mutation.WriteBuilder newInsertBuilderFromVolunteeringOpportunity() {
```

Comment:
Suggestion: There's also InsertOrUpdate for mutation builders, if that would suit your code better

Reply:
Ack, noted but does not work well if primary key is not one of form fields - thanks.